huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go

```diff
@@ -139,9 +146,57 @@ func setAuditLogPolicy(cfg *config.Config, d *schema.ResourceData, instanceId, o
 	if err != nil {
 		return fmt.Errorf("error %s DDS audit log policy: %s", operateMethod, err)
 	}
+
+	timeout := schema.TimeoutCreate
+	if operateMethod == "updating" {
+		timeout = schema.TimeoutUpdate
+	}
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"pending"},
+		Target:       []string{"success"},
+		Refresh:      ddsInstanceAuditLogPolicyRefreshFunc(setAuditLogPolicyClient, instanceId),
+		Timeout:      d.Timeout(timeout),
+		Delay:        2 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for instance (%s) audit log policy to be set successful: %s ", instanceId, err)
+	}
+
 	return nil
 }
 
+func ddsInstanceAuditLogPolicyRefreshFunc(client *golangsdk.ServiceClient, instanceID string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		opts := instances.ListInstanceOpts{
+			Id: instanceID,
+		}
+		allPages, err := instances.List(client, &opts).AllPages()
+		if err != nil {
+			return nil, "", err
+		}
+		instancesList, err := instances.ExtractInstances(allPages)
+		if err != nil {
+			return nil, "", err
+		}
+
+		if instancesList.TotalCount == 0 {
+			var instance instances.InstanceResponse
+			return instance, "", fmt.Errorf("the DDS instance has been deleted")
```

Comment:
```suggestion
			return nil, "", fmt.Errorf("the DDS instance has been deleted")
```

Reply:
done, TotalCount is the count of the result, it is more efficient